Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-da.sh
+# Description:
+#    Validate the installation and uninstallation of the Dependency Agent.
+#    Validate the enable and disable of the Dependency Agent.
+#######################################################################
+
+set -e
+set -x
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+# Get distro information
+GetDistro
+
+dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
+supported_distro_list=(redhat centos suse debian ubuntu)
+
+readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID"
+
+function test_case_cleanup(){
+    /opt/microsoft/dependency-agent/uninstall
+
+    for dir in "${dir_list[@]}"
+    do
+        rm -rf $dir
+    done
+
+    kill $watch_pid
+
+    LogMsg "Finished cleaning up test cases"
+}
+
+# Calls the cleanup function when interupted or exited
+trap test_case_cleanup SIGINT EXIT
+
+function verify_directory() {
+    failed_assertions=0
+    for dir in "${dir_list[@]}"
+    do
+        [ -d $dir ] && LogErr "Directory $dir exists." && SetTestStateFailed && failed_assertions=$[[failed_assertions+1]]
```

Comment:
Is incrementing `failed_assertions` really dependent upon `SetTestStateFailed` being successful? Why does `SetTestStateFailed` need to be called for each assertion failure when it will be done below if `failed_assertions` is non-zero?

Reply:
Changed the logic